Jenkinsfile

```diff
@@ -0,0 +1,319 @@
+           }
+           when {
+               expression {
+                   env.userInput == 'Gateway' || env.userInput == 'All'
+               }
+           }
+           steps {
+               echo "Deploy to ${TEST_PROJECT} "
+               promoteServiceSetup(params.OPENSHIFT_HOST, params.OPENSHIFT_TOKEN, 'maingateway-service',params.IMAGE_REGISTRY, params.IMAGENAMESPACE, env.destTag, params.TEST_PROJECT)    
+               promoteService(params.IMAGENAMESPACE, params.TEST_PROJECT,'maingateway-service', env.srcTag, env.destTag)
+           }
+        }
+        stage('Pushing to Test - fisuser') {
+            environment {
+                srcTag = 'latest'
+                destTag = 'promoteTest'
+                serviceName = 'fisuser-service'
+            }
+           when {
+               expression {
+                   env.userInput == 'FisUser' || env.userInput == 'All'
+               }
+           }
+            steps {
+                echo "Deploy to ${TEST_PROJECT} "
+                promoteServiceSetup(params.OPENSHIFT_HOST, params.OPENSHIFT_TOKEN, 'fisuser-service',params.IMAGE_REGISTRY, params.IMAGENAMESPACE, env.destTag, params.TEST_PROJECT)    
+                promoteService(params.IMAGENAMESPACE, params.TEST_PROJECT, 'fisuser-service', env.srcTag, env.destTag)
+            }
+        }
+        stage('Pushing to Test - fisalert') {
+            environment {
+                srcTag = 'latest'
+                destTag = 'promoteTest'
+                serviceName = 'fisalert-service'
+            }
+           when {
+               expression {
+                   env.userInput == 'FisAlert' || env.userInput == 'All'
+               }
+           }
+            steps {
+                echo "Deploy to ${TEST_PROJECT} "
+                promoteServiceSetup(params.OPENSHIFT_HOST, params.OPENSHIFT_TOKEN, 'fisalert-service',params.IMAGE_REGISTRY, params.IMAGENAMESPACE, env.destTag, params.TEST_PROJECT)
+                promoteService(params.IMAGENAMESPACE, params.TEST_PROJECT, 'fisalert-service', env.srcTag, env.destTag)
+            }
+        }
+        stage('Pushing to Test - nodejsalert') {
+            environment {
+                srcTag = 'latest'
+                destTag = 'promoteTest'
+                serviceName = 'nodejsalert-service'
+            }
+           when {
+               expression {
+                   env.userInput == 'UI' || env.userInput == 'All'
+               }
+           }
+            steps {
+                echo "Deploy to ${TEST_PROJECT} "
+                promoteServiceSetup(params.OPENSHIFT_HOST, params.OPENSHIFT_TOKEN, 'nodejsalert-ui',params.IMAGE_REGISTRY, params.IMAGENAMESPACE, env.destTag, params.TEST_PROJECT)
+                promoteService(params.IMAGENAMESPACE, params.TEST_PROJECT, 'nodejsalert-ui', env.srcTag, env.destTag)
+            }
+        }
+        stage('Wait for user to select module to push to production.') {
```

Comment:
same apply here , can make it reusable ? 
stage("pushing image into ${app_name} into ${env_name)})

Reply:
not that many stages, no need to separate them.